tests/src/core/testqgsdatadefinedsizelegend.cpp

```diff
@@ -0,0 +1,159 @@
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgstest.h"
+
+#include "qgsdatadefinedsizelegend.h"
+#include "qgsfontutils.h"
+#include "qgsrenderchecker.h"
+#include "qgssymbol.h"
+
+
+static QString _fileNameForTest( const QString &testName )
+{
+  return QDir::tempPath() + '/' + testName + ".png";
+}
+
+static bool _verifyImage( const QString &testName, QString &report )
+{
+  QgsRenderChecker checker;
+  checker.setControlPathPrefix( QStringLiteral( "data_defined_size_legend" ) );
+  checker.setControlName( "expected_" + testName );
+  checker.setRenderedImage( _fileNameForTest( testName ) );
+  checker.setSizeTolerance( 3, 3 );
+  bool equal = checker.compareImages( testName, 500 );
+  report += checker.report();
+  return equal;
+}
+
+static QgsRenderContext _createRenderContext( double mupp, double dpi, double scale )
+{
+  QgsRenderContext context;
+  context.setScaleFactor( dpi / 25.4 );
+  context.setRendererScale( scale );
+  context.setMapToPixel( QgsMapToPixel( mupp ) );
+  return context;
+}
+
+
+/** \ingroup UnitTests
+ * This is a unit test for legend rendering when using data-defined size of markers.
+ */
+class TestQgsDataDefinedSizeLegend : public QObject
+{
+    Q_OBJECT
+
+  public:
+
+  private slots:
+    void initTestCase();// will be called before the first testfunction is executed.
+    void cleanupTestCase();// will be called after the last testfunction was executed.
+
+    void testBasic();
+    void testCrowded();
+
+  private:
+    QString mReport;
+};
+
+void TestQgsDataDefinedSizeLegend::initTestCase()
+{
+  // Runs once before any tests are run
+  // init QGIS's paths - true means that all path will be inited from prefix
+  QgsApplication::init();
+  QgsApplication::initQgis();
+  QgsApplication::showSettings();
+
+  mReport += QLatin1String( "<h1>Data Defined Size Legend Tests</h1>\n" );
+}
+
+void TestQgsDataDefinedSizeLegend::cleanupTestCase()
+{
+  QString myReportFile = QDir::tempPath() + "/qgistest.html";
+  QFile myFile( myReportFile );
+  if ( myFile.open( QIODevice::WriteOnly | QIODevice::Append ) )
+  {
+    QTextStream myQTextStream( &myFile );
+    myQTextStream << mReport;
+    myFile.close();
+  }
+
+  QgsApplication::exitQgis();
+}
+
+void TestQgsDataDefinedSizeLegend::testBasic()
+{
+  QgsDataDefinedSizeLegend settings;
+  settings.setLegendType( QgsDataDefinedSizeLegend::LegendCollapsed );
+  settings.setFont( QgsFontUtils::getStandardTestFont( QStringLiteral( "Roman" ) ) );
```

Comment:
Can you use a bold font here (and bump up the font size)? it'll be more forgiving to the cross platform rendering differences